libs/common/src/main/java20/org/opensearch/common/round/BtreeSearcher.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.round;
+
+import org.opensearch.common.annotation.InternalApi;
+
+import jdk.incubator.vector.LongVector;
+import jdk.incubator.vector.Vector;
+import jdk.incubator.vector.VectorOperators;
+import jdk.incubator.vector.VectorSpecies;
+
+/**
+ * It uses vectorized B-tree search to find the round-down point.
+ *
+ * @opensearch.internal
+ */
+@InternalApi
+class BtreeSearcher implements Roundable {
```

Comment:
Can we add tests for this?

Reply:
Thanks! 

Connecting the dots, in that case, can we use the system property and lane size to verify it is BinarySearcher that is returned as part of the check (https://github.com/opensearch-project/OpenSearch/pull/11194#discussion_r1421730384)
```
assertTrue(List.of("BtreeSearcher", "BinarySearcher").contains(roundable.getClass().getSimpleName()));
```

Reply:
So we still won't be able to access the lane size without reflection as it won't be available below JDK 16 (Vector API). The other alternative is to expose `IS_BTREE_SEARCH_SUPPORTED` in the main sources as well, which is leaking details of the Java 20 implementation. Need to make one of these trade-offs. :D

Reply:
Got it! Though this is little bit of workaround, considering the trade-offs, this is a better solution. Thanks for the details.

Reply:
@reta I want to revive our previous discussion on this point. There's a possibility that the build time JDK is >= 20 but the vector lanes is < 4. In that case, the tests run with `BinarySearcher` only, potentially miss out from catching bugs.

Since we decided not to have separate test sources for Java 20 and avoided the use of reflection, do you have a better solution to handle this for tests? Ideally, I would have liked to run these SIMD tests with a random vector species (like how it was parameterized in the original implementation).

Reply:
@reta Instead of checking the number of vector lanes, I've refactored the unit-test to use reflection to construct the instance of BtreeSearcher if available. The advantage of doing this is, if the number of vector lanes (a runtime property) of the build time JDK is less than 4, the BtreeSearcher implementation is never tested. A bug in BtreeSearcher may go uncaught only to surface in production.

With this approach, all _compilable_ implementations of Roundables are always tested without complicating the build process with separate sources.

@backslasht Hope this resolves your concern too.

Reply:
@ketanv3 I would suggest a cleaner option for you, let's enrich semantics of the `opensearch.experimental.feature.simd.rounding.enabled` to:

```
opensearch.experimental.feature.simd.rounding.enabled = true | false | forced
```

with value `forced` let us ignore lanes and always use SIMD

Reply:
Thanks @ketanv3 and @reta - This looks neat.